### ISSUE TITLE
Steam.ofNullable false positive

### ISSUE THREAD

**Issue by sanyarnd** [typetools/checker-framework#3378]
```java
import java.util.Collection;
import java.util.List;
import java.util.stream.Stream;
import org.checkerframework.checker.nullness.qual.Nullable;

public class Test {

    public static void main(String[] args) {
        System.out.println(f(List.of("a", "b", "cd")));
        System.out.println(f(null));
    }

    static int f(@Nullable Collection<String> list) {

        return Stream.ofNullable(list).flatMap(Collection::stream)
                .map(String::length)
                .reduce(0, Integer::sum);
    }
}

```

Version: `Checker Framework 3.4.1`

Error message:
```
Error:(17, 48) java: [methodref.receiver.invalid] Incompatible receiver type
  Method
    @Initialized @NonNull Stream<@Initialized @NonNull String> stream(@Initialized @NonNull Collection<@Initialized @NonNull String> this) in java.util.Collection
  is not a valid method reference for
    @Initialized @NonNull Stream<? extends @Initialized @NonNull String> apply(@Initialized @NonNull Function<@Initialized @Nullable Collection<@Initialized @NonNull String>, @Initialized @NonNull Stream<? extends @Initialized @NonNull String>> this, @Initialized @Nullable Collection<@Initialized @NonNull String> p0) in java.util.function.Function
  found   : @Initialized @NonNull Collection<@Initialized @NonNull String>
  required: @Initialized @Nullable Collection<@Initialized @NonNull String>
```

**Comment by smillst** [comment]
The error is no longer issued.
